### PR TITLE
feat(Integrations: Laravel): update user login and signup events

### PR DESCRIPTION
**Pull request by Leiyks**
### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

File: appsec/tests/extension/headers_collection_14.phpt [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+All headers are collected when track_user_signup_event_automated is triggered by user
+--INI--
+extension=ddtrace.so
+datadog.appsec.log_file=/tmp/php_appsec_test.log
+datadog.appsec.log_level=debug
+datadog.appsec.enabled=1
+--ENV--
+HTTP_X_FORWARDED_FOR=7.7.7.7
+DD_TRACE_CLIENT_IP_HEADER_DISABLED=true
+HTTP_X_FORWARDED_FOR=7.7.7.6,10.0.0.1
```
Comment (cataphract) [inline appsec/tests/extension/headers_collection_14.phpt:11]: There's no point keep testing that the same headers are collected over and over. This just makes the tests difficult to read and ensures that when we modify the set of headers of we collect we need to change 20+ tests minimum.

File: src/DDTrace/Integrations/Laravel/LaravelIntegration.php [hunk 2]
```diff
@@ -44,6 +44,19 @@ public function isArtisanQueueCommand()
...
+    {
+        $allowList = ["email", "username"];
+
+        foreach ($allowList as $key) {
+            if (isset($args[$key])) {
+                return $args[$key];
+            }
+        }
+
+        return null;
```
Comment (cataphract) [inline src/DDTrace/Integrations/Laravel/LaravelIntegration.php:57]: why create a local variable and loop through it?... you can just write

```php
if (key_exists('email', $args)) {
  return $args['email'];
}
// ...
```

This would be more efficient and just plain easier to read

File: tests/Appsec/Mock.php [hunk 3]
```diff
@@ -87,74 +88,132 @@ public function getEvents(array $names = [], array $addresses = [])
...
 
 if (!function_exists('datadog\appsec\appsecMockEnabled')) {
-    function appsecMockEnabled() {
+    function appsecMockEnabled()
+    {
         return getenv('APPSEC_MOCK_ENABLED') === "true";
     }
 }
 
+if (!function_exists('datadog\appsec\track_user_login_success_event_automated')) {
```
Comment (cataphract) [inline tests/Appsec/Mock.php:97]: you should write integrations tests in appsec to ensure this is working with the actual appsec module

**Review by cataphract — CHANGES_REQUESTED** [review]

File: appsec/src/extension/tags.c [hunk 4]
```diff
@@ -918,82 +956,219 @@ static PHP_FUNCTION(datadog_appsec_track_user_signup_event)
...
-            mlog(dd_log_warning, "Unexpected empty user id");
+
+        user_login = dd_user_info_anonymize(user_login);
+        if (user_login == NULL) {
+            mlog(dd_log_debug, "Failed to anonymize user login");
             return;
         }
     }
 
+    if (user_id && ZSTR_LEN(user_id) > 0) {
```
[the remark this answers is not in the document]
Comment (Leiyks) [reply appsec/src/extension/tags.c:1002]: According to the RFC I would say that user_id is optional, however I am not sure we should add the tags if the value provided is `null` or empty.

In any case I can make it optional in the function, it would indeed make more sense.

File: src/DDTrace/Integrations/Laravel/LaravelIntegration.php (hunk 2 above)
Comment (Leiyks) [reply src/DDTrace/Integrations/Laravel/LaravelIntegration.php:57]: Yeah that's right, I wanted to just have to modify the list if we needed to add / remove some fields but that's not really needed for only two fields

File: src/DDTrace/Integrations/Laravel/LaravelIntegration.php [hunk 5]
```diff
@@ -14,9 +14,9 @@
  */
 class LaravelIntegration extends Integration
 {
-    const NAME = 'laravel';
+    public const NAME = 'laravel';
 
-    const UNNAMED_ROUTE = 'unnamed_route';
+    public const UNNAMED_ROUTE = 'unnamed_route';
```
Comment (bwoebi) [inline src/DDTrace/Integrations/Laravel/LaravelIntegration.php:19]: constant visibility only exists as of PHP 7.1. We support 7.0.

File: tests/Appsec/Mock.php [hunk 6]
```diff
@@ -87,74 +88,132 @@ public function getEvents(array $names = [], array $addresses = [])
...
         }
         $event = [
+            'userLogin' => $userLogin,
             'userId' => $userId,
             'exists' => $exists,
             'metadata' => $metadata,
-            'automated' => $automated
 
         ];
         AppsecStatus::getInstance()->addEvent($event, 'track_user_login_failure_event');
```
Comment (estringana) [inline tests/Appsec/Mock.php:150]: The name here should be `track_user_login_failure_event_automated` instead of `track_user_login_failure_event`

File: tests/Appsec/Mock.php [hunk 7]
```diff
@@ -87,74 +88,132 @@ public function getEvents(array $names = [], array $addresses = [])
...
             return;
         }
         $event = [
+            'userLogin' => $userLogin,
             'userId' => $userId,
             'metadata' => $metadata,
-            'automated' => $automated
 
         ];
         AppsecStatus::getInstance()->addEvent($event, 'track_user_signup_event');
```
Comment (estringana) [inline tests/Appsec/Mock.php:187]: Same here

File: appsec/src/extension/tags.c [hunk 8]
```diff
@@ -1095,85 +1270,150 @@ static PHP_FUNCTION(datadog_appsec_track_user_login_failure_event)
...
 
-            if (metadata != NULL && zend_array_count(metadata) > 0) {
-                metadata = NULL;
-            }
+        if (metadata && zend_array_count(metadata) > 0) {
+            metadata = NULL;
         }
     }
 
+    if (ZSTR_LEN(user_id) > 0) {
```
Comment (cataphract) [inline appsec/src/extension/tags.c:1319]: This looks like it's leaking `anon_user_id` / `anon_user_login` if `user_id` has length 0 (and `mode == user_mode_anon`)

Comment (Leiyks) [reply appsec/src/extension/tags.c:1320]: Fixed it and added some tests to make sure it's fine

**Review by cataphract — APPROVED** [review]

File: src/DDTrace/Integrations/WordPress/WordPressIntegration.php [hunk 9]
```diff
@@ -101,10 +108,11 @@ function ($par, $retval) {
                 if (\property_exists($data, 'display_name')) {
                     $metadata['name'] = $data->display_name;
                 }
-                \datadog\appsec\track_user_login_success_event(
+
+                \datadog\appsec\track_user_login_success_event_automated(
+                    $username,
                     $id,
                     $metadata,
```
Comment (PROFeNoM) [inline src/DDTrace/Integrations/WordPress/WordPressIntegration.php:115]:
```suggestion
                    $metadata
```

File: src/DDTrace/Integrations/Symfony/SymfonyIntegration.php [hunk 10]
```diff
@@ -126,23 +126,26 @@ function ($This, $scope, $args) {
...
+
                 $metadata = [];
+                $user = \method_exists($token, 'getUsername') ? $token->getUsername() : '';
 
-                \datadog\appsec\track_user_login_success_event(
-                    \method_exists($token, 'getUsername') ? $token->getUsername() : '',
+                \datadog\appsec\track_user_login_success_event_automated(
+                    $user,
+                    $user,
                     $metadata,
```
Comment (PROFeNoM) [inline src/DDTrace/Integrations/Symfony/SymfonyIntegration.php:148]:
```suggestion
                    $metadata
```

File: src/DDTrace/Integrations/Symfony/SymfonyIntegration.php [hunk 11]
```diff
@@ -189,11 +192,12 @@ function ($This, $scope, $args) {
...
 
                 $metadata = [];
+                $user = \method_exists($token, 'getUsername') ? $token->getUsername() : '';
 
-                \datadog\appsec\track_user_login_success_event(
-                    \method_exists($token, 'getUsername') ? $token->getUsername() : '',
+                \datadog\appsec\track_user_login_success_event_automated(
+                    $user,
+                    $user,
                     $metadata,
```
Comment (PROFeNoM) [inline src/DDTrace/Integrations/Symfony/SymfonyIntegration.php:200]:
```suggestion
                    $metadata
```

File: src/DDTrace/Integrations/Symfony/SymfonyIntegration.php [hunk 12]
```diff
@@ -226,17 +230,20 @@ function ($This, $scope, $args) {
...
-                \datadog\appsec\track_user_login_success_event(
-                    \method_exists($user, 'getUserIdentifier') ? $user->getUserIdentifier() : '',
+
+                $metadata = [];
+                $userIdentifier = \method_exists($user, 'getUserIdentifier') ? $user->getUserIdentifier() : '';
+
+                \datadog\appsec\track_user_login_success_event_automated(
+                    $userIdentifier,
+                    $userIdentifier,
                     $metadata,
```
Comment (PROFeNoM) [inline src/DDTrace/Integrations/Symfony/SymfonyIntegration.php:246]:
```suggestion
                    $metadata
```

File: src/DDTrace/Integrations/Laravel/LaravelIntegration.php [hunk 13]
```diff
@@ -360,24 +378,28 @@ function ($This, $scope, $args, $loginSuccess) use ($integration) {
...
+
                 if (isset($args[1]['email'])) {
                     $metadata['email'] = $args[1]['email'];
                 }
-                \datadog\appsec\track_user_login_success_event(
+
+                \datadog\appsec\track_user_login_success_event_automated(
+                    $this->getLoginFromArgs($args[1]),
                     \method_exists($args[1], 'getAuthIdentifier') ? $args[1]->getAuthIdentifier() : '',
                     $metadata,
```
Comment (PROFeNoM) [inline src/DDTrace/Integrations/Laravel/LaravelIntegration.php:402]:
```suggestion
                    $metadata
```

File: src/DDTrace/Integrations/Laravel/LaravelIntegration.php [hunk 14]
```diff
@@ -398,17 +420,19 @@ function ($This, $scope, $args) use ($integration) {
...
+
                 if (isset($args[0]['email'])) {
                     $metadata['email'] = $args[0]['email'];
                 }
 
-                \datadog\appsec\track_user_login_success_event(
+                \datadog\appsec\track_user_login_success_event_automated(
+                    $this->getLoginFromArgs($args[0]),
                     \method_exists($args[0], 'getAuthIdentifier') ? $args[0]->getAuthIdentifier() : '',
                     $metadata,
```
Comment (PROFeNoM) [inline src/DDTrace/Integrations/Laravel/LaravelIntegration.php:435]:
```suggestion
                    $metadata
```

**Review by PROFeNoM — COMMENTED** [review]
I believe that the trailing commas are the reason for the failing tests in PHP 7.2 and earlier
